cmd/sonobuoy/app/e2e.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"compress/gzip"
+	"fmt"
+	"os"
+
+	"github.com/heptio/sonobuoy/pkg/client"
+	"github.com/heptio/sonobuoy/pkg/errlog"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	show        string
+	rerunFailed bool
+)
+
+func init() {
+	cmd := &cobra.Command{
+		Use:   "e2e",
+		Short: "inspect e2e results",
+		Run:   e2es,
+		Args:  cobra.ExactArgs(1),
+	}
+	RootCmd.AddCommand(cmd)
+	AddKubeconfigFlag(&runFlags.kubecfg, cmd)
+	cmd.PersistentFlags().StringVar(&show, "show", "failed", "Defines which tests to show, options are [passed, failed (default) or all]. Cannot be combined with --rerun-failed.")
+	cmd.PersistentFlags().BoolVar(&rerunFailed, "rerun-failed", false, "Rerun the failed tests reported by the archive. The --show flag will be ignored.")
+}
+
+func e2es(cmd *cobra.Command, args []string) {
+	f, err := os.Open(args[0])
+	if err != nil {
+		errlog.LogError(errors.Wrapf(err, "could not open sonobuoy archive: %v", args[0]))
+		os.Exit(1)
+	}
+	defer f.Close()
+	// As documented, ignore show if we are doing a rerun of failed tests.
+	if rerunFailed {
+		show = "failed"
+	}
+	gzr, err := gzip.NewReader(f)
+	if err != nil {
+		errlog.LogError(errors.Wrap(err, "could not make a gzip reader"))
+		os.Exit(1)
+	}
+	defer gzr.Close()
+	sonobuoy := client.NewSonobuoyClient()
+	testCases, err := sonobuoy.GetTests(gzr, show)
+	if err != nil {
+		errlog.LogError(errors.Wrap(err, "could not get tests from archive"))
+		os.Exit(1)
+	}
+
+	// If we are not doing a rerun, print and exit.
+	if !rerunFailed {
+		fmt.Print(testCases)
+		os.Exit(0)
+	}
+
+	// Create the necessary configs for rerunning.
+	runOptions := &client.RunConfig{
+		GenConfig: client.GenConfig{
+			E2EConfig: &client.E2EConfig{
+				Focus: testCases.Focus(),
+			},
+			// Quick with a custom Focus config will work.
```

Comment:
you shouldn't need any of the options below, they should default.   If not, we should fix that.